app/controllers/buyers/applications_controller.rb

```diff
@@ -191,8 +187,18 @@ def find_provider
     @provider = current_account
   end
 
-  def find_application_plan
-    @application_plan = @provider.application_plans.find params[:cinstance][:plan_id]
+  def accessible_services
+    @accessible_services ||= current_user.accessible_services.includes(:application_plans)
+  end
+
+  helper_method :accessible_services
+
+  def accessible_plans
+    current_account.application_plans.where(issuer: accessible_services)
+  end
+
+  def plan_id
+    @plan_id ||= params.require(:cinstance).permit(:plan_id).tap { |plan_params| plan_params.require(:plan_id) }[:plan_id]
```

Comment:
Is `tap` really needed here? Does it work simply like this?

```suggestion
    @plan_id ||= params.require(:cinstance).require(:plan_id)
```

Reply:
@guicassolato same as https://github.com/3scale/porta/pull/1798#discussion_r443405932